### PR TITLE
Modify rule S6759: Restrict the scope to functional components

### PR DESCRIPTION
**Pull request by yassin-kammoun-sonarsource**
I am removing the JavaScript part of the initial RSPEC about Flow.js. Although we do support Flow.js to some extent, we limit ourselves to parsing its specific syntax. However, we don't have rule implementations that visit AST node types specific to Flow.js.

### CODE REVIEW

File: rules/S6759/javascript/rule.adoc [hunk 1]
```diff
@@ -1,82 +1,51 @@
...
-----
-type Props = {
-  +name: string,
-}
-class Welcome extends React.Component<Props> {
-  render () {
-    return <div>Hello {this.props.name}</div>;
-  }
+function Welcome(props: Props) { // Noncompliant: The component props are not read-only
+  return <div>Hello {this.props.name}</div>;
```
Comment (alexander-kamushkin-sonarsource) [inline rules/S6759/javascript/rule.adoc:16]:
```suggestion
  return <div>Hello {props.name}</div>;
```

File: rules/S6759/javascript/rule.adoc [hunk 2]
```diff
@@ -1,82 +1,51 @@
...
+interface Props {
+  name: string;
 }
-class Welcome extends React.Component<Props> {
-  render () {
-    return <div>Hello {this.props.name}</div>;
-  }
+
+function Welcome(props: Readonly<Props>) {
+  return <div>Hello {this.props.name}</div>;
```
Comment (alexander-kamushkin-sonarsource) [inline rules/S6759/javascript/rule.adoc:29]:
```suggestion
  return <div>Hello {props.name}</div>;
```

File: rules/S6759/javascript/rule.adoc [hunk 3]
```diff
@@ -1,82 +1,51 @@
...
+interface Props {
   readonly name: string;
 }
-class Welcome extends React.Component<Props> {
-  render () {
-    return <div>Hello {this.props.name}</div>;
-  }
+
+function Welcome(props: Props) {
+  return <div>Hello {this.props.name}</div>;
```
Comment (alexander-kamushkin-sonarsource) [inline rules/S6759/javascript/rule.adoc:42]:
```suggestion
  return <div>Hello {props.name}</div>;
```

**Review by alexander-kamushkin-sonarsource — CHANGES_REQUESTED** [review]

File: rules/S6759/javascript/rule.adoc (hunk 1 above)
Comment (yassin-kammoun-sonarsource) [reply rules/S6759/javascript/rule.adoc:16]: Thank you so much for catching this and the rest!

**Review by alexander-kamushkin-sonarsource — APPROVED** [review]

**Pull request merged**